ydb/public/sdk/cpp/client/ydb_persqueue_core/impl/read_session.h

```diff
@@ -1029,6 +1041,13 @@ class TSingleClusterReadSessionImpl : public NPersQueue::TEnableSelfContext<TSin
         return Log;
     }
 
+    const NTopic::ICodec* TryGetCodecImpl(ECodecAlias<UseMigrationProtocol> codecId) const {
```

Comment:
Just a thought. `TryGetSomething` function sounds like a "safe" version of `GetSomething` that returns an empty result in case of a failure. And only `GetSomething` throws an exception.